pytube/captions.py

```diff
@@ -45,12 +46,16 @@ def xml_captions(self) -> str:
     @property
     def json_captions(self) -> dict:
         """Download and parse the json caption tracks."""
-        json_captions_url = self.url.replace('fmt=srv3','fmt=json3')
+        json_captions_url = self.get_url_for_json()
```

Comment:
i'm unsure of the need for this change ... seems one line

Reply:
Yes, I have created another function, as I saw that there is no Test code for the function, I wrote a test code. For this purpose I have to make another function. Please review that